leshan-server-core/src/main/java/org/eclipse/leshan/server/client/RegistrationService.java

```diff
@@ -39,13 +39,12 @@
     Registration getByEndpoint(String endpoint);
 
     /**
-     * Returns an unmodifiable list of all registrations.
-     * 
-     * @return the registrations.
-     * @deprecated should be replace by an iterator
+     * Returns an iterator over all registrations. There are no guarantees concerning the order in which the
+     * elements are returned.
+     *
+     * @return an <tt>Iterator</tt> over registrations
      */
-    @Deprecated
-    Collection<Registration> getAllRegistrations();
+    Iterator<Registration> getAllRegistrations();
```

Comment:
What's the difference between `RegistrationService` and `RegistrationStore`? Both seem to have this method but it is unclear to me why we need both.
I also wonder whether this method is used by any other client than the demo server web UI for retrieving the list of registered devices. In that case, returning an iterator doesn't seem to be a good fit because it does not provide any support for efficient paging, does it?